compiler/qsc_linter/src/lints/ast.rs

```diff
@@ -122,3 +121,65 @@ fn precedence(expr: &qsc_ast::ast::Expr) -> u8 {
         _ => 0,
     }
 }
+
+/// Crates a lint for deprecated user-defined types declarations using `newtype`.
+impl<'compilation> AstLintPass<'compilation> for DeprecatedNewtype<'compilation> {
+    fn check_item(&self, item: &Item, buffer: &mut Vec<Lint>) {
+        if let ItemKind::Ty(_, _) = item.kind.as_ref() {
+            buffer.push(lint!(self, item.span));
+        }
+    }
+}
+
+impl DeprecatedFunctionConstructor<'_> {
+    fn resolve_item_relative_to_user_package(
+        &self,
+        item_id: &hir::ItemId,
+    ) -> (&hir::Item, &hir::Package, hir::ItemId) {
+        self.resolve_item(self.user_package_id, item_id)
+    }
```

Comment:
This is starting to look more like an HIR-related lint, maybe it should go in the lints/hir.rs file?

Reply:
Yes I agree, it is better in the HIR as it save some of the look ups and can reuse infrastructure that other HIR lints have to use.